tracer/src/Datadog.Trace/AppSec/Waf/Waf.cs

```diff
@@ -255,13 +255,18 @@ private UpdateResult Update(object arguments)
                 updated = UpdateWafAndDispose(encodedArgs);
 
                 // only if rules are provided will the waf give metrics
-                if (arguments is Dictionary<string, object> dic && dic.ContainsKey("rules"))
+                if (arguments is Dictionary<string, object> dic && dic.ContainsKey("rules") && updated.Success)
                 {
-                    TelemetryFactory.Metrics.RecordCountWafUpdates();
+                    TelemetryFactory.Metrics.RecordCountWafUpdates(Telemetry.Metrics.MetricTags.WafStatus.Success);
+                }
+                else
+                {
+                    TelemetryFactory.Metrics.RecordCountWafUpdates(Telemetry.Metrics.MetricTags.WafStatus.Error);
```

Comment:
if the waf wasn't provided with "rules", is it still considered an error? isn't it just in case `updated.Success == false`?

Reply:
Good point! After discussing it with @Anilm3, we should send the telemetry metric with success=true even if there are no rules. I have updated the code. Thanks!